AtTools/AtEDistortionModel.h

```diff
@@ -0,0 +1,90 @@
+#ifndef ATEDISTORTIONMODEL_H
+#define ATEDISTORTIONMODEL_H
+
+#include "AtSpaceChargeModel.h"
+
+#include <Rtypes.h> // for Double_t
+
+#include <fstream>
+#include <type_traits> // for add_pointer_t
+#include <unordered_map>
+
+class AtDigiPar;
+/**
+ * @brief Model for E-field distortion correction
+ *
+ *
+ *
+ *     The implemented correction is applied to every hit
+ *     Radial, transversal and longitudinal (z) corrections as a function of radial distance and z (map keys).
+ *     The Class EField mapping contains the correction values
+ */
+
+struct EFieldCorr {
+   EFieldCorr() : _zcorr(0), _radcorr(0), _tracorr(0) {}
+   EFieldCorr(Double_t zcorr, Double_t radcorr, Double_t tracorr) : _zcorr(zcorr), _radcorr(radcorr), _tracorr(tracorr)
+   {
+   }
+   inline void SetZCorr(Double_t zcorr) { _zcorr = zcorr; }
+   inline void SetRadCorr(Double_t radcorr) { _radcorr = radcorr; }
+   inline void SetTraCorr(Double_t tracorr) { _tracorr = tracorr; }
+   inline Double_t GetZCorr() { return _zcorr; }
+   inline Double_t GetRadCorr() { return _radcorr; }
+   inline Double_t GetTraCorr() { return _tracorr; }
+
+   Double_t _zcorr;
+   Double_t _radcorr;
+   Double_t _tracorr;
+};
```

Comment:
This has too much machinery around it that will keep users from using modern c++ features with structs, many of which are disabled when user-provided constructors (or default init values) are added and they are treated like classes instead of like structs. Should probably just be 
```
struct EFieldCorr {
Double_t zcorr;
Double_t radcorr;
Double_t tracorr;
};
```
(See AtPadReference for example)